src/core/mesh/qgsmeshdatasetgroupstore.h

```diff
@@ -127,8 +127,19 @@ class QgsMeshDatasetGroupStore: public QObject
     //! Constructor
     QgsMeshDatasetGroupStore( QgsMeshLayer *layer );
 
-    //! Sets the persistent mesh data provider
-    void setPersistentProvider( QgsMeshDataProvider *provider );
+    /**
+     *  Sets the persistent mesh data provider
+     *
+     * \deprecated since QGIS 3.20
+     */
+    Q_DECL_DEPRECATED void setPersistentProvider( QgsMeshDataProvider *provider );
```

Comment:
Since this isn't exposed to sip it's not part of the stable API promise. You can remove it immediately.

Reply:
Indeed the class is not in the API, but maybe more because the class is not export? (there is also the c++ API) 
And SIP_NO_FILE not necessary.

Reply:
C++ API isn't consider stable. We are free to change that whenever we need/want to

Reply:
> C++ API isn't consider stable. We are free to change that whenever we need/want to

hmmm, is there a place where it is written? C++ API also can be used (I do)... the doc here (https://qgis.org/api/3.18/api_break.html) lets make thinking that c++ API is stable.

Reply:
It was decided years ago, as an acceptable trade off between 3rd party development vs qgis development. I'm not sure if it's ever been written down formally, but it's certainly been standard practice for as long as I've been involved in qgis dev! 😂

Reply:
(the API break doc was written for pyqgis developers specifically)